torch/csrc/autograd/variable.h

```diff
@@ -42,9 +51,16 @@ struct Variable : public at::Tensor {
   inline const Variable & grad() const;
   inline       Variable & grad();
 
+  inline bool is_leaf() const;
+
   inline const std::shared_ptr<Function>& grad_fn() const;
   inline       std::shared_ptr<Function>& grad_fn();
```

Comment:
Can we make it always return a const shared pointer? I think all modifications to `grad_fn` should be through `set_history` and `rebase_history`, right?